documentation/meta/traffic/runbooks/identifying-and-blocking-traffic-anomalies.md

```diff
@@ -0,0 +1,161 @@
+# Identifying and Blocking Traffic Anomalies in Cloudflare
+
+## Objective
+
+Enable consistent and prompt responses to traffic anomalies by implementing
+procedures for identifying and blocking suspicious traffic—high request rates,
+high error volume, or scraping patterns which disrupt service—in Openverse.
+
+## When to use this runbook
+
+Use this runbook when either of these are true:
+
+1. There is a service disruption to the Openverse API or frontend.
+2. There is a marked increase in resource usage on the API or frontend ECS
+   services that continues to trend to unsustainable levels, which _could lead_
+   to a service disruption.
+
+And:
+
+1. There is a marked increase in traffic visible in our AWS dashboards or
+   Cloudflare analytics.
+
+## Requirements
+
+1. Access to the Openverse Cloudflare dashboard.
+2. Familiarity with the Cloudflare web UI, including the Analytics, Security,
+   and Web Access Firewall tabs.
+3. Time. This is an exploratory process that can take significant trial-and
+   error.
+
+## Procedure Steps
+
+### 1. Identify the timeframe of the traffic anomalies
+
+1.  Log into Cloudflare.
+2.  Select the relevant domain in the Cloudflare UI.
+3.  Go to `Analytics & Logs => Traffic` and choose the "Page Views" box in the
+    left sidebar.
+4.  Set the time range dropdown to 24 hours.
+5.  See if you can find the point where the current spike in traffic begins.
+6.  If you do not see a place where traffic stabilizes to a relatively flat
+    line, continue increasing the time range in the dropdown.
+7.  Once you find where normalized traffic starts to spike in the graph, hover
+    over the graph and **record the date and time the spike began**.
+
+### 2. Identify targetable sources of the suspicious traffic
+
+Next, we need to associate the traffic with specific sources. Essentially, we're
+trying to find unique identifiers linked to potentially malicious traffic, so we
+can block them through the Cloudflare Web Access firewall.
+
+There are three primary identifiers for targeting traffic. These are ordered
+from largest, most persistent, and most impactful, to smallest and most
+ephemeral. When possible, **blocking ASNs** is the single best way to mitigate
+malicious traffic.
+
+#### Traffic Source Identifiers
+
+- Source ASNs (Autonomous System Numbers): Unique identifiers for a group of IP
+  networks and routers under the control of a single organization, typically an
+  internet service provider or hosting company.
+
+  Typically, a single ASN makes up less than 10% of total Openverse requests in
+  a given timeframe. If this percentage is larger, it could suggest that a user
+  on the ASN is scraping Openverse or accessing Openverse programmatically in an
+  aggressive way.
```

Comment:
Slight rewording for clarity. The re-wording helps clarify for me what the difference between scraping and other programmatic usages are if they're both malicious.

```suggestion
  A single ASN usually makes up less than 10% of total Openverse requests in
  a given timeframe. If the percentage of requests from a single ASN is higher
  than 10%, it may indicate that one or more users on the ASN are scraping
  Openverse or over using/abusing expected programmatic access patterns.
```

Reply:
So much better, thanks!